crates/storage_api2/src/lib.rs

```diff
@@ -84,6 +85,32 @@ pub trait StorageApi: Send + Sync + 'static {
     ) -> impl Future<Output = Result<operation::Output>> + Send {
         async move { self.execute_ref(&operation).await }
     }
+
+    /// Pulls all data stored in a WCN Database within the specified
+    /// `keyrange`.
+    ///
+    /// Intended to be implemented for WCN Databases and WCN Replicas, but
+    /// not WCN Coordinators.
+    ///
+    /// Same [`KeyspaceVersion`] validation applies as for the regular
+    /// [`Operation`]s.
+    fn pull_data(
```

Comment:
I would call these two `read` and `write` instead of pull/push.

Reply:
FWIW I couldn't understand the difference between pull and push without reading the doc comments. With read/write I'd be able to immediately tell what these methods do.

Reply:
I prefer read and write too in this case, as @heilhead pull/push isnt obvious without domain knowledege, I'd argue that even with domain knowledge it's easy to forget what these represent

Reply:
Interesting, I thought pull/push was a widely used concept. And we have also been using it before for the same purpose in this codebase.

To me `read`/`write` usually means something that works with `Read`/`Write` or `AsyncRead`/`AsyncWrite`, which are purely raw byte interfaces.
And pull push is very similar, but instead typed.